### PR TITLE
Ci: start testing py310-dev

### PR DESCRIPTION
**Pull request by fangchenli**
This might not work now. Let's see.

### CODE REVIEW

**Comment by jreback** [comment]
nice,ping when ready (also happy to merge if we can xfail this build while in-progress)

**Comment by fangchenli** [comment]
pytest error due to https://github.com/pytest-dev/pytest/issues/8539

File: .github/workflows/python-dev.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
```
Comment (fangchenli) [inline .github/workflows/python-dev.yml:17]: btw, this is the proper way to use checkout@v2 to get all tags

**Comment by jreback** [comment]
looks great. can you rename this workflow similar to our others but just add dev e.g. actions-310-dev or similar. ping on green.

**Comment by fangchenli** [comment]
testing will timeout if using the xdist or cov plugin. not sure why

**Comment by fangchenli** [comment]
If we use coverage directly, the test would start running. But there were some errors in json tests and it freeze there.

```
pandas/tests/io/json/test_compression.py ......EEEEE....................ssss.........
pandas/tests/io/json/test_deprecated_kwargs.py ..
pandas/tests/io/json/test_json_table_schema.py .............................................................................................x...x...x...x...x...x...x...x...x...x.......................
pandas/tests/io/json/test_normalize.py ................................................
Error: The operation was canceled.
```

**Comment by jreback** [comment]
thanks @fangchenli very nice!

can you open some issues about the failures (and tag with 3.10, going to create it now). thanks!

**Pull request merged**